src/util/gateway/index.ts

```diff
@@ -80,3 +85,17 @@ export interface DiscoveryQueueMatchPacket extends GatewayPacket {
 		channel: APIDMChannel;
 	};
 }
+
+export interface ClientTypingPacket extends GatewayPacket {
+	type: GatewayPacketType.ClientTyping;
+	data: {
+		channel: APIChannel;
+	};
+}
+
+export interface GatewayTypingPacket extends GatewayPacket {
+	type: GatewayPacketType.GatewayTyping;
+	data: {
+		user: APIUser;
```

Comment:
Could you turn this into userID since it seems wasteful and inefficient to fetch and send the entire user every few seconds.

This also needs to store the channelID - since otherwise the client won't know which channel they are typing in.